### PR TITLE
Fix double encoding XML in storage queues.

### PR DESCRIPTION
**Pull request by rakshith91**
Resolves Azure#8035

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client.py [hunk 1]
```diff
@@ -106,11 +106,11 @@ def __init__(
         if 'message_encode_policy' in kwargs:
             self._config.message_encode_policy = kwargs['message_encode_policy'] or NoEncodePolicy()
         else:
-            self._config.message_encode_policy = TextXMLEncodePolicy()
+            self._config.message_encode_policy = NoEncodePolicy()
```
Comment (lmazuel) [inline sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client.py]: Could be simplified to 
```python
self._config.message_encode_policy = kwargs.get('message_encode_policy', NoEncodePolicy())
```

File: sdk/storage/azure-storage-queue/tests/recordings/test_queue_encodings.test_message_text_xml.yaml [hunk 2]
```diff
@@ -88,24 +88,24 @@ interactions:
...
   response:
     body:
-      string: "\uFEFF<?xml version=\"1.0\" encoding=\"utf-8\"?><QueueMessagesList><QueueMessage><MessageId>ff2a9e11-0842-4dc9-a69f-daac1634e864</MessageId><InsertionTime>Wed,
-        16 Oct 2019 06:53:15 GMT</InsertionTime><ExpirationTime>Wed, 23 Oct 2019 06:53:15
-        GMT</ExpirationTime><PopReceipt>AgAAAAMAAAAAAAAAz1ctdO6D1QE=</PopReceipt><TimeNextVisible>Wed,
-        16 Oct 2019 06:53:45 GMT</TimeNextVisible><DequeueCount>1</DequeueCount><MessageText>&amp;lt;message1&amp;gt;</MessageText></QueueMessage></QueueMessagesList>"
+      string: "\uFEFF<?xml version=\"1.0\" encoding=\"utf-8\"?><QueueMessagesList><QueueMessage><MessageId>1fce4157-6682-4a5b-8887-6e2f27f6639d</MessageId><InsertionTime>Tue,
+        22 Oct 2019 21:32:40 GMT</InsertionTime><ExpirationTime>Tue, 29 Oct 2019 21:32:40
+        GMT</ExpirationTime><PopReceipt>AgAAAAMAAAAAAAAA2+kMTSCJ1QE=</PopReceipt><TimeNextVisible>Tue,
+        22 Oct 2019 21:33:10 GMT</TimeNextVisible><DequeueCount>1</DequeueCount><MessageText>&lt;message1&gt;</MessageText></QueueMessage></QueueMessagesList>"
```
Comment (lmazuel) [inline sdk/storage/azure-storage-queue/tests/recordings/test_queue_encodings.test_message_text_xml.yaml]: That's what I wanted to see :)

File: sdk/storage/azure-storage-queue/azure/storage/queue/_message_encoding.py [hunk 3]
```diff
@@ -133,33 +134,12 @@ def decode(self, content, response):
...
-    def decode(self, content, response):
-        return xml_unescape(content)
-
-
 class NoEncodePolicy(MessageEncodePolicy):
     """Bypass any message content encoding."""
 
     def encode(self, content):
+        if isinstance(content, six.binary_type) and sys.version_info > (3,):
+            raise ValueError("Message content must not be bytes.")
```
Comment (lmazuel) [inline sdk/storage/azure-storage-queue/azure/storage/queue/_message_encoding.py]: Raise TypeError, that was the previous behavior anyway (see the update in tests you did)

File: sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client.py (hunk 1 above)
Comment (rakshith91) [reply sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client.py]: hmm, None and NoEncodepolicy are not same.

Comment (johanste) [reply sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client.py]: @rakshith91, what is the semantic difference (for this method) between explicitly passing in `None` as the `message_encode_policy` and using `NoEncodePolicy`?

**Comment by johanste** [comment]
I would not expect that many recordings to be changed in semantically meaningful ways. Have you verified that all the changes are expected?

File: sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client.py (hunk 1 above)
Comment (rakshith91) [reply sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client.py]: We don't expose NoEncodePolicy() . Not passing anything or passing None explicitly will both  have NoEncodePolicy

**Comment by rakshith91** [comment]
> I would not expect that many recordings to be changed in semantically meaningful ways. Have you verified that all the changes are expected?

yes..most recordings just had their timestamp changed...the ones involving send message have the correct message now.

File: sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client.py (hunk 1 above)
Comment (rakshith91) [reply sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client.py]: https://github.com/rakshith91/azure-sdk-for-python/blob/text_encoding_queue/sdk/storage/azure-storage-queue/tests/test_queue_encodings.py#L196

https://github.com/rakshith91/azure-sdk-for-python/blob/text_encoding_queue/sdk/storage/azure-storage-queue/tests/test_queue_encodings.py#L65

These two tests will explain it better...the first one has a None passed..the second has nothing...it's still NoEncodePolicy in both cases.

File: sdk/storage/azure-storage-queue/azure/storage/queue/_message_encoding.py [hunk 4]
```diff
@@ -133,33 +132,12 @@ def decode(self, content, response):
...
-    def decode(self, content, response):
-        return xml_unescape(content)
-
-
 class NoEncodePolicy(MessageEncodePolicy):
     """Bypass any message content encoding."""
 
     def encode(self, content):
+        if isinstance(content, six.binary_type) and sys.version_info > (3,):
+            raise TypeError("Message content must not be bytes.")
```
Comment (lmazuel) [inline sdk/storage/azure-storage-queue/azure/storage/queue/_message_encoding.py]: Thinking: should the exception suggst to use the Base64 encoder to guide people? Something like;

"If you really wants to send bytes, you can the Base64Encoder"

Comment (rakshith91) [reply sdk/storage/azure-storage-queue/azure/storage/queue/_message_encoding.py]: +1

File: sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client.py (hunk 1 above)
Comment (lmazuel) [reply sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client.py]: With the line I suggest it's still NoEncodePolicy in both case. I don't understand your point? Both are equivalent, but one is taking 4 lines :)

**Pull request merged**